src/reduxTypes.js

```diff
@@ -330,31 +330,31 @@ export type UsersState = User[];
  * identify which subtrees are persisted and which are not.
  */
 export type GlobalState = {|
-  accounts: AccountsState,
-  alertWords: AlertWordsState,
-  caughtUp: CaughtUpState,
-  drafts: DraftsState,
-  fetching: FetchingState,
-  flags: FlagsState,
-  messages: MessagesState,
-  migrations: MigrationsState,
-  mute: MuteState,
-  mutedUsers: MutedUsersState,
-  narrows: NarrowsState,
-  outbox: OutboxState,
-  pmConversations: PmConversationsState,
-  presence: PresenceState,
-  realm: RealmState,
-  session: SessionState,
-  settings: SettingsState,
-  streams: StreamsState,
-  subscriptions: SubscriptionsState,
-  topics: TopicsState,
-  typing: TypingState,
-  unread: UnreadState,
-  userGroups: UserGroupsState,
-  userStatus: UserStatusState,
-  users: UsersState,
+  +accounts: AccountsState,
+  +alertWords: AlertWordsState,
+  +caughtUp: CaughtUpState,
+  +drafts: DraftsState,
+  +fetching: FetchingState,
+  +flags: FlagsState,
+  +messages: MessagesState,
+  +migrations: MigrationsState,
+  +mute: MuteState,
+  +mutedUsers: MutedUsersState,
+  +narrows: NarrowsState,
+  +outbox: OutboxState,
+  +pmConversations: PmConversationsState,
+  +presence: PresenceState,
+  +realm: RealmState,
+  +session: SessionState,
+  +settings: SettingsState,
+  +streams: StreamsState,
+  +subscriptions: SubscriptionsState,
+  +topics: TopicsState,
+  +typing: TypingState,
+  +unread: UnreadState,
+  +userGroups: UserGroupsState,
+  +userStatus: UserStatusState,
+  +users: UsersState,
```

Comment:
Cann you add more explanation to the commit message about why you're making these covariant? I read the flow docs on it (which it'd be nice to link to, btw), but that doesn't tell me why you want covariance.

Reply:
Yeah, I agree this could use more explanation. I've raised a question on CZO [here](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Make.20.60GlobalState.60.20covariant.20in.20its.20properties.3F/near/1185670) as I'm not sure how best to condense the reason down from 130 Flow errors.

Reply:
I think this is resolved by using `$ReadOnly`, which does the same thing but is clearer about what it does; I've done that in my new revision.